tools/itch5bookdepth.cpp

```diff
@@ -0,0 +1,145 @@
+#include <jb/itch5/compute_book_depth.hpp>
+#include <jb/itch5/process_iostream.hpp>
+#include <jb/book_depth_statistics.hpp>
+#include <jb/fileio.hpp>
+#include <jb/log.hpp>
+
+#include <iostream>
+#include <stdexcept>
+#include <unordered_map>
+
+#include <chrono>
+
+/**
+ * Compute ITCH5 Depth of Book statistics.
+ *
+ * Generate statistics per symbol and aggregated.
+ * See https://github.com/GFariasR/jaybeams/wiki/ITCH5-Depth-of-Book-StatisticsProject
+ * for design and implementation details.
+ */
+namespace {
+
+class config : public jb::config_object {
+ public:
+  config();
+  config_object_constructors(config);
+
+  void validate() const override;
+
+  jb::config_attribute<config,std::string> input_file;
+  jb::config_attribute<config,std::string> output_file;
+  jb::config_attribute<config,jb::log::config> log;
+  jb::config_attribute<config,jb::book_depth_statistics::config> stats;
+  jb::config_attribute<config,jb::book_depth_statistics::config> symbol_stats;
+  jb::config_attribute<config,bool> enable_symbol_stats;
+};
+
+} // anonymous namespace
+
+typedef std::chrono::high_resolution_clock high_resolution_t;
+typedef std::chrono::microseconds microseconds_t;
+
+int main(int argc, char* argv[]) try {
+  config cfg;
+  cfg.load_overrides(
+      argc, argv, std::string("itch5bookdepth.yaml"), "JB_ROOT");
+  jb::log::init(cfg.log());
+
+  boost::iostreams::filtering_istream in;
+  jb::open_input_file(in, cfg.input_file());
+
+  boost::iostreams::filtering_ostream out;
+  jb::open_output_file(out, cfg.output_file());
+
+  std::map<jb::itch5::stock_t, jb::book_depth_statistics> per_symbol;
+  jb::book_depth_statistics stats(cfg.stats());
+
+  auto cb = [&](
+      jb::itch5::compute_book_depth::time_point recv_ts,
+      jb::itch5::message_header const& header,
+      jb::itch5::stock_t const& stock,
+      jb::itch5::book_depth_t const& book_depth) {
+    auto pl = std::chrono::steady_clock::now() - recv_ts;
```

Comment:
In addition, if I change a class file (compute_book_depth.cpp for instance)
I got this error:
      ar: `u' modifier ignored since `D' is the default (see `U')

While, if I run it again, changing itch5bookdepth.cpp (for instance) there is no error reported

Is that right?